### PR TITLE
Secure things down a bit more

### PR DESCRIPTION
**Pull request by Mekle001**
There were quite a few places that hadn't yet been updated to check for the user being logged in. I think I've caught most of those.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,27 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  protected
+
+  # Confirms the correct user.
+  def correct_user
+    @user = User.find(params[:id])
+    redirect_to(root_url) unless current_user?(@user)
```
Comment (NepoKama) [inline app/controllers/application_controller.rb:10]: This is checking to see if the user matches the user we think is logged in, yes?  What can cause those not to match?  Should this force a logout?

**Review by NepoKama — APPROVED** [review]
Are we going to want additional user tiers?  At this point any user can edit any coin, correct?  I'm thinking we want to limit most users on how much they can edit and create a new role (Editor?) that can edit records but it not an admin.

**Comment by Mekle001** [comment]
I very much agree. This was the basic Auth system the tutorial built. Seems reasonably straightforward to extend it. It's probably time to build a task list to work down.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Mekle001) [reply app/controllers/application_controller.rb:10]: This scenario is hit for viewing the user profile page. It's not worthy of a log out, just a "you can't see this, if it's not you"

**Pull request merged**